### PR TITLE
add waits for idle in add product to order

### PR DESCRIPTION
**Pull request by rrennick**
### Changes proposed in this Pull Request:

This PR adds two network waits to the `addProductToOrder`. If you re-run the merchant tests several times the order search test will start to consistently fail. I believe the reason for that is that many of the merchant tests create and trash products. Once there is a significant number of products in the DB, the product search lookup slows and the OK button is clicked with the select list empty.

See #29666 .

### How to test the changes in this Pull Request:

1. Verify CI tests
2. Run E2E tests locally

### Changelog entry

N/A

### CODE REVIEW

**Comment by rrennick** [comment]
@zhongruige This one is ready for review. Could you try it both with screenshots enabled and disabled?

**Comment by zhongruige** [comment]
@rrennick It seems like these tests are failing pretty consistently for me with both screenshots enabled and disabled, was curious if you're seeing the same as well, even though it passed fine in the CI:

Screen Shot 2021-04-21 at 2 03 49 PM

File: tests/e2e/utils/src/components.js
Comment (zhongruige) [inline tests/e2e/utils/src/components.js:458]: Not that we need to change this here, but this made me wonder if it's easier to pass in the API name (e.g. `fixed_cart`, `fixed_product` etc) compared with the UI name, which could change.

File: tests/e2e/utils/src/components.js
Comment (zhongruige) [inline tests/e2e/utils/src/components.js:474]: Great idea updating this to use the API 👍

**Comment by zhongruige** [comment]
Also of note, the variation products and general settings updates seemed to fail (with the screenshot for the variable products failure showing the settings page), but I wasn't sure if that was intended to be addressed on this PR or not.

Screen Shot 2021-04-21 at 3 04 39 PM

Screen Shot 2021-04-21 at 3 04 59 PM

File: tests/e2e/utils/src/components.js
Comment (rrennick) [reply tests/e2e/utils/src/components.js:458]: > Not that we need to change this here, but this made me wonder if it's easier to pass in the API name

We could make that the `default:` which would allow updating the parameter but keep it backward compatible.

> It seems like these tests are failing pretty consistently for me 

Did you run `npm run build:packages` on this branch?

Comment (zhongruige) [reply tests/e2e/utils/src/components.js:458]: > Did you run npm run build:packages on this branch?

I did when I initially tested (also just did another run-through to check for sure) and I am still seeing those failures, unfortunately.

Comment (zhongruige) [reply tests/e2e/utils/src/components.js:458]: Just to follow up on the above, it seems it was some flakiness I was hitting locally. After making the changes on this PR https://github.com/woocommerce/woocommerce/pull/29743 I no longer was hitting the repeated failures above.

Comment (rrennick) [reply tests/e2e/utils/src/components.js:455]: > Not that we need to change this here, but this made me wonder if it's easier to pass in the API name (e.g. fixed_cart, fixed_product etc) compared with the UI name, which could change, especially with the changes to use the API below.

@zhongruige I updated to include support for this. This is ready for another review.

File: tests/e2e/core-tests/specs/merchant/wp-admin-order-status-filters.test.js
Comment (zhongruige) [inline tests/e2e/core-tests/specs/merchant/wp-admin-order-status-filters.test.js:60]: I wonder if we should look at using `Batch update orders` to create orders like this (I believe the API will allow passing in a status, if I recall correctly) to not only avoid flakiness and needing to up the timeout, but also make the test more efficient.

File: tests/e2e/core-tests/specs/merchant/wp-admin-product-new.test.js
Comment (zhongruige) [inline tests/e2e/core-tests/specs/merchant/wp-admin-product-new.test.js:89]: I like this. I wonder if we also want to start thinking about adding in a logout step after all the tests run, e.g.

```
afterAll(async () => {
	await merchant.logout();
});
```

Just to prevent future flakiness and make the tests a bit more atomic and clean up after themselves. We don't need to change this here, but just a thought.

**Review by zhongruige — DISMISSED** [review]
Thanks for the work on this @rrennick! The changes look good, and these ran great for me locally. Also verified the update tests passed locally as well:

```
  WooCommerce Orders > Apply coupon
    ✓ can apply a coupon (698ms)
    ✓ can remove a coupon (273ms)
```

```
  WooCommerce Orders > Filter Orders by Status
    ✓ should filter by Pending payment (2317ms)
    ✓ should filter by Processing (3370ms)
    ✓ should filter by On hold (3302ms)
    ✓ should filter by Completed (3363ms)
    ✓ should filter by Cancelled (2306ms)
    ✓ should filter by Refunded (2345ms)
    ✓ should filter by Failed (2341ms)
    ✓ should filter by All (2350ms)
```

```
  Products > Edit Product
    ✓ can edit a product and save the changes (4733ms
```

```
  Add New Variable Product Page
    ✓ can create variable product (1640ms)
    ✓ can create set variable product attributes (2557ms)
    ✓ can create variable product variations (910ms)
    ✓ can create variation attributes (5438ms)
```

**Comment by zhongruige** [comment]
@rrennick Just noticed the merge conflicts after approving, once these are resolved we should be good to merge this in.

**Review by zhongruige — APPROVED** [review]
Re-ran the tests after merging in `trunk` and everything passed :+1: and on the first run, which showing the flakiness we've been fixing is working well, too. Nice work on all this @rrennick!

```
Test Suites: 5 passed, 5 total
Tests:       3 skipped, 160 passed, 163 total
Snapshots:   0 total
Time:        709.856s
Ran all test suites.
Jest exit code: 0
```

This PR is good to go! :shipit:

File: tests/e2e/core-tests/specs/merchant/wp-admin-order-status-filters.test.js
Comment (rrennick) [reply tests/e2e/core-tests/specs/merchant/wp-admin-order-status-filters.test.js:60]: Yes, this should be added as one of the tasks to complete after #27873.

File: tests/e2e/core-tests/specs/merchant/wp-admin-product-new.test.js
Comment (rrennick) [reply tests/e2e/core-tests/specs/merchant/wp-admin-product-new.test.js:89]: Good suggestion. I created #29796.

File: tests/e2e/core-tests/specs/merchant/wp-admin-order-status-filters.test.js
Comment (zhongruige) [reply tests/e2e/core-tests/specs/merchant/wp-admin-order-status-filters.test.js:60]: Sounds good--I created https://github.com/woocommerce/woocommerce/issues/29797 to follow up on updating these tests to use the API once it's added.

**Pull request merged**
